### ISSUE TITLE
angel_framework:Compatibility issue with combinator version 1.0.0

### ISSUE THREAD

**Issue by Antolius** [angel-dart/roadmap#19]
# Compatibility issue with combinator version 1.0.0
*Occurs in angel_framework*

## Versions:
Angel framework version 1.1.3 (and lower).
Transitive [combinator](https://github.com/thosakwe/combinator) version 1.0.0 (released on May 30th and automatically caught by the `combinator: ">=1.0.0-beta <2.0.0"` range in angel_framework's `pubspec.yaml`)

## Steps to reproduce error:
Execute the following program
```dart
import 'package:angel_framework/angel_framework.dart';

main() {
  new Angel().get('/:id', () => 'OK');
}
```

For sake of completeness, here's my `pubspec.yaml` as well:
```yaml
name: "angel_test"
dependencies: 
  angel_framework: 1.1.3
environment: 
  sdk: 2.0.0-dev.59.0
```

## Stack trace
```
Unhandled exception:
NoSuchMethodError: Class '_RegExpParser' has no instance method '__parse'.
Receiver: Instance of '_RegExpParser'
Tried calling: __parse(Instance of 'ParseArgs')
#0      Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
#1      Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#2      _Opt.__parse (package:combinator/src/combinator/opt.dart:12:25)
#3      Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#4      _Chain.__parse (package:combinator/src/combinator/chain.dart:51:27)
#5      Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#6      _Map.__parse (package:combinator/src/combinator/map.dart:11:25)
#7      Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#8      _Any._parse (package:combinator/src/combinator/any.dart:36:27)
#9      _Chain.__parse (package:combinator/src/combinator/chain.dart:51:27)
#10     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#11     _Map.__parse (package:combinator/src/combinator/map.dart:11:25)
#12     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#13     _Map.__parse (package:combinator/src/combinator/map.dart:11:25)
#14     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#15     _Chain.__parse (package:combinator/src/combinator/chain.dart:51:27)
#16     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#17     _Index.__parse (package:combinator/src/combinator/index.dart:11:25)
#18     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#19     _CastDynamic.__parse (package:combinator/src/combinator/cast.dart:41:25)
#20     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#21     _Cast.__parse (package:combinator/src/combinator/cast.dart:10:25)
#22     Parser._parse (package:combinator/src/combinator/combinator.dart:82:18)
#23     Parser.parse (package:combinator/src/combinator/combinator.dart:91:12)
#24     new Route (package:angel_route/src/route.dart:15:14)
#25     Router.addRoute (package:angel_route/src/router.dart:77:23)
#26     Routable.addRoute (package:angel_framework/src/core/routable.dart:89:26)
#27     Angel.addRoute (package:angel_framework/src/core/server.dart:178:26)
#28     Router.get (package:angel_route/src/router.dart:387:12)
#29     main (file://~/angel_test/server.dart:4:15)
#30     _startIsolate.<anonymous closure> (dart:isolate-patch/dart:isolate/isolate_patch.dart:279)
#31     _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:165)
```

## How urgently do I need a fix?
Not really urgent, since overriding the combinator version to `combinator: 1.0.0-beta+7` in my own pubspec will mitigate the issue. However, would be nice to resolve the problem.

**Comment by thosakwe** [comment]
Thanks for pointing this out, I've noticed it as well. Working on a fix. The package in question is `angel_route`.

**Comment by thosakwe** [comment]
Also, thank you for the *incredibly detailed* error report. Makes it a lot easier to hunt this down!

**Comment by thosakwe** [comment]
Fixed as of version `2.0.6` of `angel_route`!
